packages/FormElement/src/FormElement.js

```diff
@@ -71,85 +71,91 @@ function FormElement(props) {
 
   function renderFooter() {
-    if (isFooterInserted) return;
-
-    isFooterInserted = true;
-
     if (hasError) {
-      return extratedChildren["FormElement.Error"];
+      return React.cloneElement(extractedChildren["FormElement.Error"], {
+        ariaErrorId,
+      });
     }
 
-    if (extratedChildren["FormElement.Description"]) {
-      return React.cloneElement(extratedChildren["FormElement.Description"], {
+    if (extractedChildren["FormElement.Description"]) {
+      return React.cloneElement(extractedChildren["FormElement.Description"], {
         ariaDescriptionId,
       });
     }
 
     return null;
   }
 
+  function renderInstructions() {
+    if (extractedChildren["FormElement.Instructions"]) {
+      return React.cloneElement(extractedChildren["FormElement.Instructions"], {
+        ariaInstructionsId,
+      });
+    }
+  }
+
+  const nativeChildTypes = ["input", "textarea", "select"];
+
+  const renderFormElementChild = (child, index) => (
+    <div key={`child-key_${index}`} css={formElementChildStyle}>
+      {child}
+    </div>
+  );
+
+  const ariaDescribedByIds = `${ariaErrorId} ${ariaInstructionsId} ${ariaDescriptionId}`;
+
   return (
     <div css={formElementStyles} isInline={isInline} size={size} isDisabled={isDisabled} {...moreProps}>
       <Label
         hasOptionalLabel={hasOptionalLabel}
         hasRequiredLabel={hasRequiredLabel}
-        help={extratedChildren["FormElement.Help"]}
+        help={extractedChildren["FormElement.Help"]}
         id={inputId}
         isInline={isInline}
         isVisuallyHidden={isLabelVisuallyHidden}
         label={label}
       />
-
-      {extratedChildren.children.map(child => {
-        if (React.isValidElement(child)) {
-          const extendedProps = isString(child.type)
-            ? {
-                "aria-describedby": ariaDescriptionId,
-                id: inputId,
-                disabled: isDisabled,
-                readOnly: isReadOnly,
-              }
-            : {
-                "aria-describedby": ariaDescriptionId,
-                hasError,
-                id: inputId,
-                isDisabled,
-                isReadOnly,
-                size,
-              };
-          const clonedChild = React.cloneElement(child, extendedProps);
-
-          return (
-            <React.Fragment key={child.key}>
-              {isInline ? (
-                <div css={inlineContainerStyles}>
-                  {clonedChild}
-                  {renderFooter()}
-                </div>
-              ) : (
-                <React.Fragment>
-                  {clonedChild}
-                  {renderFooter()}
-                </React.Fragment>
-              )}
-            </React.Fragment>
-          );
-        }
-
-        return child;
-      })}
+      <div css={isInline ? inlineContainerStyles : null}>
+        {renderInstructions()}
+        {extractedChildren.children.map((child, index) => {
+          if (React.isValidElement(child)) {
```

Comment:
I think the logic here is backwards? 
If `isValidElement` then we should pass `isDisabled`, not `disabled`, for example.

Reply:
Maybe we can add a set of helper functions like this? https://stackoverflow.com/questions/33199959/how-to-detect-a-react-component-vs-a-react-element sometimes `React.isValidElement` isn't enough, that's why I have to add `isString(child.type)` checking before

Reply:
Oh As the code was previously I figure the `isString` method was evaluating if it was a react component, and that `React.isValidElement` was checking that it was just an element of some kind, native or otherwise. I'll adjust it 👍

Reply:
Actually i believe all of the children will be react components at this point as they have been extracted via the extractChildren function, so I am removing the check for React.isValidElement as seems redundant. I am checking the type however to determine if it is a native element or not.